model.basin.add() in model with existing nodes will cause AssertionError next-time model.node_table() is is accessed

While not to explicitly check available node_id in model with existing nodes, I run this code:

```
        geometry = basin_to_point(basin_polygon=basin_polygon, tolerance=10)
        node = Node(
            meta_categorie="bergend",
            geometry=geometry,
        )
        data = [basin.Profile(level=[0, 1], area=[5, 100]), basin.Area(geometry=[basin_polygon])]
        basin_node = model.basin.add(node=node, tables=data)
```

Inspecting the `model.basin.node.df` I notice node_id =1 has been assigned. Does not exist in the basin-table, but does exist as `FlowBoundary`:
![image](https://github.com/user-attachments/assets/427473d8-9539-493c-ab94-336e253c2f10)

Thereby accessing `model.node_table()` will raise an `AssertionError: node_id must be unique`:
![image](https://github.com/user-attachments/assets/19ad2bfe-3922-47a9-ad64-39351a1ac5cc)

--- a/python/ribasim/ribasim/geometry/edge.py
+++ b/python/ribasim/ribasim/geometry/edge.py
@@ -10,7 +10,7 @@ from numpy.typing import NDArray
 from pandera.dtypes import Int32
 from pandera.typing import Index, Series
 from pandera.typing.geopandas import GeoDataFrame, GeoSeries
-from pydantic import NonNegativeInt, PrivateAttr
+from pydantic import NonNegativeInt, PrivateAttr, model_validator
 from shapely.geometry import LineString, MultiLineString, Point
 
 from ribasim.input_base import SpatialTableModel
@@ -59,6 +59,13 @@ class EdgeTable(SpatialTableModel[EdgeSchema]):
     """Defines the connections between nodes."""
 
     _used_edge_ids: UsedIDs = PrivateAttr(default_factory=UsedIDs)
+
+    @model_validator(mode="after")
+    def _update_used_ids(self) -> "EdgeTable":
+        if self.df is not None and len(self.df.index) > 0:
+            self._used_edge_ids.node_ids.update(self.df.index)
+            self._used_edge_ids.max_node_id = self.df.index.max()
+        return self
 
     def add(
         self,

--- a/python/ribasim/ribasim/model.py
+++ b/python/ribasim/ribasim/model.py
@@ -124,6 +124,16 @@ class Model(FileModel):
         self.edge.df.set_geometry("geometry", inplace=True, crs=self.crs)
         return self
 
+    @model_validator(mode="after")
+    def _update_used_ids(self) -> "EdgeTable":
+        # Only update the used node IDs if we read from a database
+        if "database" in context_file_loading.get():
+            df = self.node_table().df
+            if len(df.index) > 0:
+                self._used_node_ids.node_ids.update(df.index)
+                self._used_node_ids.max_node_id = df.index.max()
+        return self
+
     @field_serializer("input_dir", "results_dir")
     def _serialize_path(self, path: Path) -> str:
         return str(path)
@@ -224,7 +234,11 @@ class Model(FileModel):
     def node_table(self) -> NodeTable:
         """Compute the full sorted NodeTable from all node types."""
         df_chunks = [node.node.df for node in self._nodes()]
-        df = pd.concat(df_chunks)
+        df = (
+            pd.concat(df_chunks)
+            if df_chunks
+            else pd.DataFrame(index=pd.Index([], name="node_id"))
+        )
         node_table = NodeTable(df=df)
         node_table.sort()
         assert node_table.df is not None

--- a/python/ribasim/ribasim/model.py
+++ b/python/ribasim/ribasim/model.py
@@ -125,10 +125,11 @@ class Model(FileModel):
         return self
 
     @model_validator(mode="after")
-    def _update_used_ids(self) -> "EdgeTable":
+    def _update_used_ids(self) -> "Model":
         # Only update the used node IDs if we read from a database
         if "database" in context_file_loading.get():
             df = self.node_table().df
+            assert df is not None
             if len(df.index) > 0:
                 self._used_node_ids.node_ids.update(df.index)
                 self._used_node_ids.max_node_id = df.index.max()

--- a/python/ribasim/tests/test_io.py
+++ b/python/ribasim/tests/test_io.py
@@ -202,6 +202,19 @@ def test_node_autoincrement():
     assert nbasin.node_id == 101
 
 
+def test_node_autoincrement_existing_model(basic, tmp_path):
+    model = basic
+
+    model.write(tmp_path / "ribasim.toml")
+    nmodel = Model.read(tmp_path / "ribasim.toml")
+
+    assert nmodel._used_node_ids.max_node_id == 17
+    assert nmodel._used_node_ids.node_ids == set(range(1, 18)) - {13}
+
+    assert nmodel.edge._used_edge_ids.max_node_id == 16
+    assert nmodel.edge._used_edge_ids.node_ids == set(range(1, 17))
+
+
 def test_node_empty_geometry():
     model = Model(
         starttime="2020-01-01",

